flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/source/PostgresDialect.java

```diff
@@ -214,6 +218,21 @@ public JdbcSourceFetchTaskContext createFetchTaskContext(
         return new PostgresSourceFetchTaskContext(taskSourceConfig, this);
     }
 
+    @Override
+    public FetchTask.Context createFetchTaskContext(
+            SourceSplitBase sourceSplitBase, JdbcSourceConfig taskSourceConfig, boolean reuse) {
+        if (!reuse) {
+            return createFetchTaskContext(sourceSplitBase, taskSourceConfig);
+        }
+        if (this.taskSourceConfig == null
+                || this.fetchTaskContext == null
+                || !this.taskSourceConfig.equals(taskSourceConfig)) {
+            this.taskSourceConfig = taskSourceConfig;
+            this.fetchTaskContext = createFetchTaskContext(sourceSplitBase, taskSourceConfig);
```

Comment:
It seem that fetchTaskContext will be shared, what happens if the first one invoke FetchTask.Context#close to close the connection and then another task use this fetchTaskContext?